3.2ListOfContacts/ListOfContacts/ContactStorage.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace ListOfContacts
+{
+    public sealed class ContactStorage
+    {
+
+        private readonly List<string> keys = new List<string>()
+        {
+            "English",
+            "Ukrainian",
+            "#",
+            "Number"
+        };
+
+        private Dictionary<string, List<Contact>> phoneBook = new Dictionary<string, List<Contact>>();
+        public ContactStorage()
+        {
+            phoneBook.Add(keys[0], new List<Contact>());
+            phoneBook.Add(keys[1], new List<Contact>());
+            phoneBook.Add(keys[2], new List<Contact>());
+            phoneBook.Add(keys[3], new List<Contact>());
+        }
+
+        public void AddContact(string nameAndSurname, string phoneNumber)
+        {
+            Contact contact = new Contact()
+            {
+                NameAndSurname = nameAndSurname,
+                PhoneNumber = phoneNumber
+            };
+
+            try
+            {
+                if (IsEnglish(nameAndSurname) == true && Regex.IsMatch(phoneNumber, @"^\u002b\d{12}") && phoneNumber.StartsWith("+44"))
+                {
+                    phoneBook[keys[0]].Add(contact);
+                }
+
+                else if (IsUkrainian(nameAndSurname) == true && Regex.IsMatch(phoneNumber, @"^\u002b\d{12}") && phoneNumber.StartsWith("+38"))
+                {
+                    phoneBook[keys[1]].Add(contact);
+                }
+
+                else if (IsEnglish(nameAndSurname) == false && IsUkrainian(nameAndSurname) == false && Regex.IsMatch(phoneNumber, @"^\u002b\d{12}") && phoneNumber.StartsWith("+"))
```

Comment:
IsEnglish(nameAndSurname) == false replace with !IsEnglish(nameAndSurname)

Reply:
I think, I fixed this.